torch/distributed/_fsdp/fully_sharded_data_parallel.py

```diff
@@ -177,7 +177,7 @@ def __init__(
         # shard any leftover parameters.
         params = []
         for param_name, param in module.named_parameters():
-            if not hasattr(param, "_is_sharded"):
+            if not isinstance(param, FlatParameter):
```

Comment:
let us still check 'if not hasattr(p, "_is_sharded")', which is more strict to require parameters are sharded

Reply:
comment above, let's discuss there.